From: SJ Park <sj@kernel.org>
Subject: mm/damon/core: hide private damon_probe fields
Date: Tue, 14 Jul 2026 07:35:41 -0700

'filters' and 'list' fields of damon_probe are intended to be used by only
the DAMON core layer.  However, those are mistakenly not marked as
private.  Mark as private.

Link: https://lore.kernel.org/20260714143544.101305-9-sj@kernel.org
Signed-off-by: SJ Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/damon.h |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/include/linux/damon.h~mm-damon-core-hide-private-damon_probe-fields
+++ a/include/linux/damon.h
@@ -775,12 +775,13 @@ struct damon_filter {
  * struct damon_probe - Data region attribute probe.
  *
  * @weight:	Relative priority of the attribute for this probe.
- * @filters:	Filters for assessing if a given region is for this probe.
- * @list:	Siblings list.
  */
 struct damon_probe {
 	unsigned int weight;
+/* private: */
+	/* Filters for assessing if a given region is for this probe. */
 	struct list_head filters;
+	/* Siblings list. */
 	struct list_head list;
 };
 
_
